internal/dag/gatewayapi_processor.go

```diff
@@ -114,7 +114,9 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 				p.Errorf("error validating routes against Listener.Routes.Selector: %s", err)
 			}
 
-			if selMatches && nsMatches {
+			// If all the match criteria for this HTTPRoute match the Gateway, then add
+			// the route to the set of matchingRoutes.
+			if selMatches && nsMatches && p.gatewayMatches(route) {
```

Comment:
the `RouteGateways` type mentions:

```
RouteGateways defines which Gateways will be able to use a route. If this field results in preventing the selection of a Route by a Gateway, an “Admitted” condition with a status of false must be set for the Gateway on that Route.
```

Seems like we should set status of "Admitted=false" on Routes that are selected by a Gateway but the Route does not "allow" the Gateway to select it

See https://gateway-api.sigs.k8s.io/references/spec/#networking.x-k8s.io/v1alpha1.RouteGateways

Reply:
Hmm good catch. We'll need to see if they are defined now to know if they matched by default vs matching the selectors. Let me write up some more tests.

Reply:
Ok just pushed a new commit @sunjayBhatia. =)